src/components/common/CheckWallet/index.tsx

```diff
@@ -46,15 +48,21 @@ const CheckWallet = ({
     if (!wallet) {
       return Message.WalletNotConnected
     }
+
     if (isUndeployedSafe && !allowUndeployedSafe) {
       return Message.SafeNotActivated
     }
 
     if (!allowNonOwner && !isSafeOwner && !isProposer && (!isOnlySpendingLimit || !allowSpendingLimit)) {
       return Message.NotSafeOwner
     }
+
+    if (!allowProposer && isProposer) {
```

Comment:
Is it possible to be proposer and owner at the same time?
e.g. by adding an address as proposer and then later on adding the same address as owner?

This could lead to false warnings in that case when `allowProposer` is `false` and `isProposer` and `isOwner` is `true`.

Reply:
Good point! Technically, it is possible to be owner and proposer although we added validation to prevent adding owners as proposers. Still someone could do this via other means so lets extend the condition to `!allowProposer && isProposer && !isSafeOwner`